engine/CMakeLists.txt

```diff
@@ -1265,8 +1308,70 @@ ELSEIF (CMAKE_SYSTEM_NAME STREQUAL Linux)
         SET(CPACK_RPM_PACKAGE_LICENSE "GPLv3") # See ../LICENSE
         SET(CPACK_RPM_PACKAGE_URL "https://www.vega-strike.org")
-        # Should detect whether SuSe or RH/CentOS/Fedora as deps may change
-        SET(CPACK_RPM_PACKAGE_REQUIRES "python2, libjpeg, libpng, freeglut, gtk2, libvorbis, openal, SDL_mixer")
+        SET(CPACK_RPM_PACKAGE_NAME "Vega-Strike")
+
+        # Figure out the filename
+        IF (LINUX_CODENAME)
+            SET(VEGA_STRIKE_LINUX_VERSION_STR "${LINUX_CODENAME}")
+        ELSE (LINUX_CODENAME)
+            SET(VEGA_STRIKE_LINUX_VERSION_STR "${LINUX_VERSION_ID}")
+        ENDIF (LINUX_CODENAME)
+
+        SET(CPACK_PACKAGE_FILE_NAME "${CPACK_RPM_PACKAGE_NAME}_${VEGA_STRIKE_PACKAGE_VERSION_STR}-${VEGA_STRIKE_PYTHON_VERSION_STR}-${LINUX_ID}-${VEGA_STRIKE_LINUX_VERSION_STR}_x86_64")
+
+        # Detect whether SuSe or RH/CentOS/Fedora as deps may change
+        IF (LINUX_ID STREQUAL opensuse-leap)
+            IF (USE_PYTHON_3)
+                SET(CPACK_RPM_PACKAGE_REQUIRES "libboost_python-py3-1_66_0, libpython3_6m1_0")
+            ELSE (USE_PYTHON_3)
+                SET(CPACK_RPM_PACKAGE_REQUIRES "libboost_python-py2_7-1_66_0, libpython2_7-1_0")
+            ENDIF (USE_PYTHON_3)
+            SET(CPACK_RPM_PACKAGE_REQUIRES "${CPACK_RPM_PACKAGE_REQUIRES}, libjpeg-turbo, libpng16-16, libglut3, libgtk-3-0, libvorbis0, libopenal0, libSDL-1_2-0, libglvnd, expat, libboost_log1_66_0, libboost_system1_66_0, libboost_filesystem1_66_0, libboost_thread1_66_0, libboost_regex1_66_0, libboost_chrono1_66_0, libboost_atomic1_66_0")
+        ELSEIF (LINUX_ID STREQUAL fedora AND LINUX_VERSION_ID STREQUAL "32")
+            IF (USE_PYTHON_3)
+                SET(CPACK_RPM_PACKAGE_REQUIRES "python38, boost-python3")       # python39
+            ELSE (USE_PYTHON_3)
+                MESSAGE(AUTHOR_WARNING "!! boost-python2 does not appear to be available on this platform.")
+                SET(CPACK_RPM_PACKAGE_REQUIRES "python27")
+            ENDIF (USE_PYTHON_3)
+            SET(CPACK_RPM_PACKAGE_REQUIRES "${CPACK_RPM_PACKAGE_REQUIRES}, libjpeg-turbo, libpng, freeglut, gtk3, libvorbis, openal-soft, SDL, libglvnd, expat, boost-log, boost-system, boost-filesystem, boost-thread, boost-regex, boost-chrono, boost-atomic")
+        ELSEIF (LINUX_ID STREQUAL fedora AND LINUX_VERSION_ID STREQUAL "31")    # Not yet tested
+            IF (USE_PYTHON_3)
+                SET(CPACK_RPM_PACKAGE_REQUIRES "python38, boost-python3")
+            ELSE (USE_PYTHON_3)
+                SET(CPACK_RPM_PACKAGE_REQUIRES "python26, boost-python2")
+            ENDIF (USE_PYTHON_3)
+            SET(CPACK_RPM_PACKAGE_REQUIRES "${CPACK_RPM_PACKAGE_REQUIRES}, libjpeg-turbo, libpng, freeglut, gtk3, libvorbis, openal-soft, SDL, libglvnd, expat, boost-log, boost-system, boost-filesystem, boost-thread, boost-regex, boost-chrono, boost-atomic")
+        ELSEIF (LINUX_ID STREQUAL fedora AND LINUX_VERSION_ID STREQUAL "30")    # Not yet tested
```

Comment:
Q: is there some way to combine the Fedora sections? Are the package names different between each release? I find that hard to be the case. It would be better if we could do:
```
if fedora:
   if version >= X:
   else Y =< version < X:
   else version <Y:
...
```

Reply:
The Python versions are different. The other package names are probably the same.

I'll try Fedora 30 and 31 as well, then get back to you.

Reply:
OK. Tested and working on Fedora 30, 31, and 32. Let me know what you think.